mineinabyss-features/src/main/kotlin/com/mineinabyss/features/gondolas/GondolasListener.kt

```diff
@@ -0,0 +1,86 @@
+package com.mineinabyss.features.gondolas
+
+import com.mineinabyss.components.gondolas.Gondola
+import com.mineinabyss.components.gondolas.UnlockedGondolas
+import com.mineinabyss.geary.papermc.tracking.entities.toGeary
+import com.mineinabyss.idofront.messaging.error
+import org.bukkit.entity.Player
+import org.bukkit.event.Listener
+import org.bukkit.event.EventHandler
+import org.bukkit.event.player.PlayerMoveEvent
+import java.util.UUID
+import com.mineinabyss.idofront.textcomponents.miniMsg
+
+class GondolasListener : Listener {
+    private val playerZoneEntry = mutableMapOf<UUID, Pair<String, Long>>()
+    private val justWarped = mutableSetOf<UUID>()
+    private val lastErrorTime = mutableMapOf<UUID, Long>()
+    private val errorCooldown = 5000
+
+    @EventHandler
+    fun PlayerMoveEvent.onPlayerMove() {
+        if (!hasExplicitlyChangedBlock()) return
+
+        val gondolas = LoadedGondolas.loaded
+        val now = System.currentTimeMillis()
+
+        val nearbyGondolaData =
+            gondolas.entries.firstNotNullOfOrNull { (id, gondola) ->
+                val type = getClosestGondolaType(gondola, player.location)
+                if (type != GondolaType.NONE) Triple(
+                    id,
+                    gondola,
+                    type
+                ) else null
+            }
+
+        if (nearbyGondolaData == null) {
+            playerZoneEntry.remove(player.uniqueId)
+            justWarped.remove(player.uniqueId)
+            return
+        }
+
+        val (id, gondola, type) = nearbyGondolaData
```

Comment:
```kt
val (id, gondola, type) = gondolas.entries.firstNotNullOfOrNull { (id, gondola) ->
    getClosestGondolaType(gondola, player.location).takeIf { it != GondolaType.NONE }?.let { 
        Triple(id, gondola, it) 
    }
} ?: run { 
    playerZoneEntry.remove(player.uniqueId)
    justWarped.remove(player.uniqueId)
    return 
}
```

This should instead be a custom GondolaData data-class not a triple though, which is what getClosestGondolaType would return
removes alot of "bloat" here